core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCacheFactory.java

```diff
@@ -26,9 +26,14 @@
 
 /**
  * A factory for {@link ZooCache} instances.
+ * <p>
+ * Implementation note: We are using this "cache" to track all the instances that have been created,
```

Comment:
The word "cache" here is confusing since what it refers to is not an instance of any kind of cache, but we are dealing with ZooCache objects in this class. It would be better to refer to it directly,

```suggestion
 * Implementation note: We are using the instances map to track all the instances that have been created,
```

Otherwise, these changes are good.